scripts/thyra-installer.py

```diff
@@ -0,0 +1,132 @@
+                            universal_newlines=decodeBinary)
+    stdout, stderr = process.communicate()
+
+    if stderr != None and stderr != "" and errorChecking == True:
+        print("Error encountered while executing : " + str(command) + " :\n", stderr)
+        os._exit(-1)
+    return stdout
+
+def setupDNS():
+    commandOutput = executeOSCommandOrFile("wmic nic where \"netenabled=true\" get netconnectionID", True)
+
+    networkAdpatersNames = list(filter(None, commandOutput.split('\n')))
+    networkAdpatersNames.pop(0)
+    networkAdpatersNames = [name.strip() for name in networkAdpatersNames]
+
+    for name in networkAdpatersNames:
+        executeOSCommandOrFile("NETSH interface ipv4 set dnsservers " + name + " static 127.0.0.1 primary", True)
+        executeOSCommandOrFile("NETSH interface ipv6 set dnsservers " + name + " static ::1 primary", True)
+    try:
+        os.remove(DEFAULT_ACRYLIC_PATH + "\\" +  ACRYLIC_DNS_PROXY_FILENAME)
+    except OSError as err:
+        printErrorAndExit(err)
+
+def configureAcrylic():
+    print("Configuring Acrylic...")
+    f = open(DEFAULT_ACRYLIC_PATH + "\\" + ACRYLIC_HOST_FILE, "r+")
+    if f.read().find("127.0.0.1 *.massa") == -1:
+        f.write("\n127.0.0.1 *.massa")
+    f.close()
+
+    executeOSCommandOrFile("NET STOP AcrylicDNSProxySvc", True)
+    executeOSCommandOrFile("NET START AcrylicDNSProxySvc", True)
+
+def setupMkCerts():
+    if os.path.isdir(THYRA_CONFIG_FOLDER_PATH) is False:
+        try:
+            os.mkdir(THYRA_CONFIG_FOLDER_PATH)
+            os.mkdir(THYRA_CONFIG_FOLDER_PATH + "\\certs")
+        except OSError as err:
+            printErrorAndExit(err)
+
+    downloadFile(MKCERT_URL, MKCERT_FILENAME)
+    executeOSCommandOrFile([os.getcwd() + "\\" + MKCERT_FILENAME, "--install"], False, False)
+    executeOSCommandOrFile([
+        os.getcwd() + "\\" + MKCERT_FILENAME,
+        "--cert-file",
+        CERTIFICATIONS_FOLDER + "\\cert.pem",
+        "--key-file",
+        CERTIFICATIONS_FOLDER  + "\\cert-key.pem",
+        "my.massa"], False, False)
+    try:
+        os.remove(MKCERT_FILENAME)
+    except OSError as err:
+        printErrorAndExit(err)
+    print("MKcert and HTTPS certificates successfully setup")
+
+def isAdmin():
+    return ctypes.windll.shell32.IsUserAnAdmin() != 0
+
+def printErrorAndExit(error):
+    print(error)
+    os.system("pause")
+    os._exit(-1)
+
+def main():
+    if isAdmin() == False:
+        printErrorAndExit("Couldn't detect admin rights. Please execute this script as an administator.")
+    downloadFile(THYRA_URL, THYRA_FILENAME)
+    if os.path.exists(DEFAULT_ACRYLIC_PATH):
+        print("Acrylic DNS Proxy is already installed")
+    else:
+        downloadFile(ACRYLIC_DNS_PROXY_URL, ACRYLIC_DNS_PROXY_FILENAME)
+        unzipAcrylic()
+        executeOSCommandOrFile(DEFAULT_ACRYLIC_PATH + "\InstallAcrylicService.bat", True)
+        setupDNS()
+    configureAcrylic()
+    setupMkCerts()
```

Comment:
What if we aleady have a CA certificate ? If it's safe, we can live with that !

Reply:
maybe you can check the return code of mkcert ?
 https://stackoverflow.com/questions/334879/how-do-i-get-the-application-exit-code-from-a-windows-command-line

Reply:
I ended up checking the return code, which is 0 as expected.